tests/AssetBank/FileUploaderTest.php

```diff
@@ -18,269 +22,390 @@ protected function setUp()
+
+        $fileId = 'testing-file_id';
+        $fileChunk = '';
+        $brandId = 'testBrandId';
+        if ($file = fopen($filePath, 'rb')) {
+            $fileChunk = fread($file, self::CHUNK_SIZE);
+        }
+        $mediaId = 'test-media-id';
+
+        // Prepare upload.
+        $mockOauthHandler
+            ->expects($this->at(0))
+            ->method('sendRequestAsync')
+            ->with(...self::getPrepareRequest())
+            ->will($this->returnValue(self::getPrepareResponse()));
 
-        // Poll status of upload.
+        // Upload in chunks.
         $mockOauthHandler
-            ->expects($this->at(4))
+            ->expects($this->at(1))
+            ->method('sendRequestAsync')
+            ->with(...self::getUploadChunksRequest($fileId, 0, $fileChunk))
+            ->will($this->returnValue(self::getUploadChunksResponse()));
+
+        // Finalises the upload.
+        $mockOauthHandler
+            ->expects($this->at(2))
             ->method('sendRequestAsync')
-            ->with(...self::getPollStatusRequest())
-            ->will($this->returnValue(self::getPollStatusResponse()));
+            ->with(...self::getFinaliseApiRequest(
+                $fileId,
+                $filePath,
+                filesize($filePath),
+                1.0,
+                hash("sha256", $fileChunk)
+            ))
+            ->will($this->returnValue(self::getFinaliseApiResponse()));
 
         // Save media asset.
         $mockOauthHandler
-            ->expects($this->at(5))
+            ->expects($this->at(3))
             ->method('sendRequestAsync')
-            ->with(...self::getSaveMediaRequest($filePath))
+            ->with(...self::getSaveMediaRequestWithMediaId($fileId, $filePath, $brandId, $mediaId))
             ->will($this->returnValue(new FulfilledPromise('DONE')));
 
         // Start a new FileUploader instance with our mockHandlers.
-        $fileUploader = new FileUploader($mockOauthHandler, $mockAwsHandler);
+        $fileUploader = new FileUploader($mockOauthHandler);
         $fileUpload = $fileUploader->uploadFile(array(
-            'filePath' => $filePath
+            'filePath' => $filePath,
+            'brandId' => $brandId,
+            'mediaId' => $mediaId
         ));
 
-        $fileUploadRes = $fileUpload->wait();
-
         $this->assertNotNull($fileUpload);
-        $this->assertEquals($fileUploadRes, 'DONE');
+        $this->assertEquals($fileUpload,  'DONE');
     }
 
     /**
-     * Builds a valid init upload request.
+     * Tests if the exception is thrown under FileUploader::saveMediaAsync,
+     *  when a invalid brandId is specified.
+     *
+     * The order it tests is:
+     *      1. Prepare upload
+     *      2. Upload file in chunks
+     *      3. Finalise the upload 
+     *      4. Save media asset
+     *
+     * @covers \Bynder\Api\Impl\Upload\FileUploader::uploadFile()
+     * @throws Exception
+     */
+    public function testInvalidBrandId()
```

Comment:
This should only happen when you upload a new asset (so when not passing a media ID to the save request).

Furthermore, there seems to be a lot of duplicate code with the previous test, so would be good to put the common code in a reusable function.